lib/vector-core/src/schema/definition.rs

```diff
@@ -87,7 +87,7 @@ impl Definition {
     ) -> Self {
         Self {
             event_kind,
-            metadata_kind: Kind::object(Collection::empty()),
+            metadata_kind: Kind::object(Collection::any()),
```

Comment:
@fuchsnj I have had to change this since if `schema_enabled` is off we copy over the meanings, but not the kinds. To ensure the meanings that point to metadata are still valid this needs to be `any`.